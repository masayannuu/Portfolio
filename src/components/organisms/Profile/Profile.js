import React from 'react'
import styled from 'styled-components'
import { Image } from 'reactbulma'
import InViewMonitor from 'react-inview-monitor'

import Headline from '../../atoms/Title/Headline'
import StyledSection from '../../atoms/Section/StyledSection'
import Icon from '../../../../public/icon.png'

const StyledParagraph = styled.div`
  font-feature-settings: 'palt';
  letter-spacing: 0.1em;
  -webkit-animation-duration: 2s;
  -moz-animation-duration: 2s;
  animation-duration: 2s;
  font-size: 0.9em;
`

const StyledDiv = styled.div`
  @media screen and (min-width: 600px) {
    width: 593px;
  }
`

const StyledHistoryDiv = styled.div`
  margin-top: 5em;
  font-size: 0.8em;
`

const StyledImage = styled(Image)`
  -webkit-animation-duration: 1s;
  -moz-animation-duration: 1s;
  animation-duration: 1s;
  @media screen and (min-width: 300px) {
    width: 100%;
  }
  @media screen and (min-width: 600px) {
    width: 367px;
    float: right;
  }
`

const StyledAnker = styled.a`
  color: #4a4a4a;
  &:hover {
    color: #a2c4c7;
  }
`

const Profile = () => (
  <StyledSection id="profile">
    <Headline>| PROFILE</Headline>
    <div className="columns">
      <StyledDiv className="column">
        <InViewMonitor classNameNotInView="is-invisible" classNameInView="animated slideInUp">
          <StyledParagraph>
            <p>1993年2月20日生まれ。広島出身。</p>
            <p>
              バンド活動を通じてクリエイターに興味を持つ。音楽以外でも自分で構造やスキームを設計し作ることに喜びを見出し、社会人へ。
            </p>
            <p>
              1年目でアイデアソンからの事業化で挫折し、ECサイト運営を担当。プライベートでの勉強をきっかけに2年目からプログラマーに転向。
            </p>
            <p>
              年間3,500億円以上の受注・在庫連携を扱うECプラットフォーム上で、システム設計の上流から実装・運用の下流まで幅広く実践と経験を積む。
            </p>
            <p>
              プログラムでもビジネススキームでも美しい構造にフェチる。自分で創れるようになるべくして行動中。何かと夢中になる事が多く、マイペースとよく言われる。
            </p>
            <StyledHistoryDiv>
              <p>
                <b>2011~2014 Add (Ba.)</b>
              </p>
              <p>
                <b>
                  2013~
                  <StyledAnker href="http://usagibunnyboy.com/" target="_blank">
                    usagibunnyboy(ウサギバニーボーイ)
                  </StyledAnker>
                  (Ba.)
                </b>
              </p>
              <p>
                <b>
                  2015~
                  <StyledAnker href="http://hamee.co.jp/" target="_blank">
                    Hamee inc
                  </StyledAnker>
                  , EC shop marketing → Programer
                </b>
              </p>
            </StyledHistoryDiv>
          </StyledParagraph>
        </InViewMonitor>
      </StyledDiv>
      <div className="column">
        <InViewMonitor classNameNotInView="is-invisible" classNameInView="animated slideInRight">
          <StyledImage src={Icon} />
        </InViewMonitor>
      </div>
    </div>
  </StyledSection>
)

export default Profile
