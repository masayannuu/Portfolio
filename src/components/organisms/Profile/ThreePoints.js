import React from 'react'
import styled from 'styled-components'
import { Image } from 'reactbulma'
import InViewMonitor from 'react-inview-monitor'

import SubHeading from '../../atoms/Title/SubHeading'
import StyledSection from '../../atoms/Section/StyledSection'
import Icon from '../../../../public/three-sibe-back.png'

const StyledDiv = styled.div`
  @media screen and (max-width: 768px) {
    width: 100%;
  }
  @media screen and (min-width: 769px) {
    width: 593px;
  }
  @media screen and (min-width: 1408px) {
    width: 100%;
  }
  font-feature-settings: 'palt';
  letter-spacing: 0.1em;
  -webkit-animation-duration: 2s;
  -moz-animation-duration: 2s;
  animation-duration: 2s;
  font-size: 1rem;
`

const StyledImage = styled(Image)`
  @media screen and (max-width: 768px) {
    width: 100%;
  }
  @media screen and (min-width: 769px) {
    position: -webkit-sticky;
    position: sticky;
    top: 50px;
    width: 367px;
    height: 593px;
  }
  @media screen and (min-width: 1408px) {
    position: -webkit-sticky;
    position: sticky;
    top: 50px;
    width: 50%;
    height: 593px;
  }
`

const StyledAnker = styled.a`
  color: #4a4a4a;
  &:hover {
    color: #a2c4c7;
  }
`

const StyledDl = styled.dl`
  margin-bottom: 1rem;
`

const ThreePoints = () => (
  <StyledSection id="ThreePoints">
    <SubHeading>3-CLUTURES</SubHeading>
    <div className="columns">
      <StyledDiv className="column">
        <InViewMonitor classNameNotInView="vis-hidden" classNameInView="animated slideInLeft">
          <StyledImage src={Icon} />
        </InViewMonitor>
      </StyledDiv>
      <StyledDiv className="column tk-a-otf-gothic-bbb-pr6n">
        <InViewMonitor classNameNotInView="vis-hidden" classNameInView="animated slideInUp">
          <StyledDl>
            <dt>
              <b>1. 毎日読書をしている</b>
            </dt>
            <dd>本や論文は好奇心と知識の源。行動と思考を補助してくれる叡智。</dd>
            <dd>
              漫画は
              <StyledAnker
                href="https://ja.wikipedia.org/wiki/%E4%B9%9D%E4%BA%95%E8%AB%92%E5%AD%90"
                target="_blank"
              >
                九井諒子先生
              </StyledAnker>
              の「
              <StyledAnker
                href="https://www.amazon.co.jp/dp/B00E59A03O/ref=dp-kindle-redirect?_encoding=UTF8&btkr=1"
                target="_blank"
              >
                ひきだしにテラリウム
              </StyledAnker>
              」が好き。
            </dd>
          </StyledDl>
          <StyledDl>
            <dt>
              <b>2. 構造思考で見つめる</b>
            </dt>
            <dd>システム思考フェチ。図解も好き。</dd>
            <dd>幾何学な設計と、それが寂びていく風景も好き。</dd>
          </StyledDl>
          <StyledDl>
            <dt>
              <b>3. スノーボードが好き</b>
            </dt>
            <dd>10歳離れた兄とのコミュニケーション手段から始まり、見事にハマった。</dd>
            <dd>バックカントリーで見た蔵王山の樹氷に心打たれている。</dd>
          </StyledDl>
        </InViewMonitor>
      </StyledDiv>
    </div>
  </StyledSection>
)

export default ThreePoints
