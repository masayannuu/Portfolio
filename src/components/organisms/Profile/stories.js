import React from 'react'
import { storiesOf } from '@storybook/react'

import Profile from './Profile'

storiesOf('Organisms.Profile', module).add('Profile', () => <Profile />)
