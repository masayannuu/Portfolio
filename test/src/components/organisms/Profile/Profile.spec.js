import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Profile from '../../../../../src/components/organisms/Profile/Profile'

describe('Profile Profile Molecules', () => {
  it('should render Profile', () => {
    const wrapper = Enzyme.shallow(<Profile />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
