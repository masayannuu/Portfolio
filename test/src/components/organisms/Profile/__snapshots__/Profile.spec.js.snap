// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Profile Molecules should render Profile 1`] = `
<StyledSection
  id="profile"
>
  <Headline>
    | PROFILE
  </Headline>
  <div
    className="columns"
  >
    <Profile__StyledDiv
      className="column"
    >
      <InviewMonitor
        childPropsNotInView={Object {}}
        classNameInView="animated slideInUp"
        classNameNotInView="vis-hidden"
        intoViewMargin="-20%"
        useInviewMonitor={[Function]}
      >
        <Profile__StyledParagraph>
          <p>
            1993年2月20日生まれ。広島出身。
          </p>
          <p>
            バンド活動を通じてクリエイターに興味を持つ。音楽以外でも自分で構造やスキームを設計し作ることに喜びを見出し、社会人へ。
          </p>
          <p>
            新卒1年目でアイデアソンからの事業化で挫折し、ECサイト運営を担当。事業化挫折を発端に2年目からプログラマーへ転向。
          </p>
          <p>
            年間3,500億円以上の受注・在庫連携を扱うECプラットフォーム上で、ラッキーなことにシステム設計の上流から実装・運用の下流まで幅広く実践と経験を積む。
          </p>
          <p>
            2018年から社長直下の新規事業開発に参画し、IoT製品を開発しリリース。反省点が多かったPJ体制をきっかけにIssue駆動や仮説・検証の大事さを説く。開発体制を変えるべく、サーバTのリーダーとして新規開発中。
          </p>
          <p>
            プログラムでもビジネススキームでも美しい構造にフェチる。自分で創れるようになるべくして行動中。何かと夢中になる事が多く、マイペースとよく言われる。
          </p>
          <Profile__StyledHistoryDiv>
            <p>
              <b>
                2011~2015: add (Ba.)
              </b>
            </p>
            <p>
              <b>
                2013~2015
                <Profile__StyledAnker
                  href="http://usagibunnyboy.com/"
                  target="_blank"
                >
                  : usagibunnyboy(ウサギバニーボーイ)
                </Profile__StyledAnker>
                (Ba.)
              </b>
            </p>
            <p>
              <b>
                2015~
                <Profile__StyledAnker
                  href="http://hamee.co.jp/"
                  target="_blank"
                >
                  : Hamee inc
                </Profile__StyledAnker>
                , EC marketing → ECプラットフォーム Programer
              </b>
            </p>
            <p>
              <b>
                2018~
                <Profile__StyledAnker
                  href="http://hamee.co.jp/"
                  target="_blank"
                >
                  : Hamee inc
                </Profile__StyledAnker>
                , IoT部 Programer
              </b>
            </p>
          </Profile__StyledHistoryDiv>
        </Profile__StyledParagraph>
      </InviewMonitor>
    </Profile__StyledDiv>
    <div
      className="column"
    >
      <InviewMonitor
        childPropsNotInView={Object {}}
        classNameInView="animated slideInRight"
        classNameNotInView="vis-hidden"
        intoViewMargin="-20%"
        useInviewMonitor={[Function]}
      >
        <Profile__StyledImage
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </InviewMonitor>
    </div>
  </div>
</StyledSection>
`;
